FreeOrion couldn't be compiled on some architectures

Bug Report
==========
<!--
Please fill in a a meaningful and concise title in the field above.
It helps us to classify the issue even  before reading the body and handling the
issue a bit better.

Examples:
  * Great: "Unable to select fleet in galaxy map"
  * Good: "Unable to click fleet"
  * Not really okay: "Issue with fleets"
  * Horrible: "HELP!!!"
-->


Environment
-----------
<!--
You don't need to provide the information in section when they are not useful in
the context of this issue.  For example a crashing hosted server may not need
information about the graphic card, but maybe depend on the operating system you
use.  When in doubt provide all information you have available.

You can find the FreeOrion version number in the lower right corner of the game
main screen.

When stating the operating system also note the version of the operating system
so, `Windows 8 Pro` instead of `Windows` or `Mac OSX Mountain Lion` instead of
`OSX`.
-->

* **FreeOrion Version**: fd3ef5f
* **Operating System**: Ubuntu
* **Fetched as** <!-- delete lines below that don't apply -->
  * Compiled from source


Description
-----------
<!--
Add a meaningful description of the bug you encountered.

If a screenshot or image helps to describe the issue content feel free to create
one.

Also attach the log files the game creates.  The log files The log files can be
found on
* Windows: %APPDATA%\FreeOrion
* MacOSX: ~/Library/"Application Support"/FreeOrion
* Linux: ${XDG_DATA_HOME:-~/.local/share}/freeorion
-->

```
[  0%] Building CXX object GG/CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o
cd "/<<PKGBUILDDIR>>/obj-aarch64-linux-gnu/GG" && /usr/bin/c++ -DBINPATH=\"/usr/bin\" -DBOOST_ALL_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_GIL_IO_ENABLE_GRAY_ALPHA -DBOOST_GIL_USES_MP11 -DBOOST_REGEX_DYN_LINK -DENABLE_BINRELOC -DFONS_USE_FREETYPE -DFREEORION_LINUX -DGiGiHeader -DGiGi_EXPORTS -DSHAREPATH=\"/usr/share\" -I"/<<PKGBUILDDIR>>/GG" -I"/<<PKGBUILDDIR>>/obj-aarch64-linux-gnu/GG" -I/usr/include/freetype2 -I"/<<PKGBUILDDIR>>/GG/Boost::boost" -g -O2 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -O2 -g -DNDEBUG -fPIC -fvisibility=hidden -Wall -std=c++17 -MD -MT GG/CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o -MF CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o.d -o CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o -c "/<<PKGBUILDDIR>>/GG/src/dialogs/ColorDlg.cpp"
In file included from /<<PKGBUILDDIR>>/GG/GG/Button.h:22,
                 from /<<PKGBUILDDIR>>/GG/GG/dialogs/ColorDlg.h:18,
                 from /<<PKGBUILDDIR>>/GG/src/dialogs/ColorDlg.cpp:10:
/<<PKGBUILDDIR>>/GG/GG/Base.h:130:16: error: enumerator value ‘-1’ is outside the range of underlying type ‘char’
  130 |     WR_NONE = -1,
      |                ^
/<<PKGBUILDDIR>>/GG/GG/Enum.h:307:9: note: in definition of macro ‘GG_ENUM’
  307 |         __VA_ARGS__                                                                     \
      |         ^~~~~~~~~~~
make[3]: *** [GG/CMakeFiles/GiGi.dir/build.make:79: GG/CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o] Error 1

make[3]: Entering directory '/<<PKGBUILDDIR>>/obj-powerpc64le-linux-gnu'
[  0%] Building CXX object GG/CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o
cd "/<<PKGBUILDDIR>>/obj-powerpc64le-linux-gnu/GG" && /usr/bin/c++ -DBINPATH=\"/usr/bin\" -DBOOST_ALL_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_GIL_IO_ENABLE_GRAY_ALPHA -DBOOST_GIL_USES_MP11 -DBOOST_REGEX_DYN_LINK -DENABLE_BINRELOC -DFONS_USE_FREETYPE -DFREEORION_LINUX -DGiGiHeader -DGiGi_EXPORTS -DSHAREPATH=\"/usr/share\" -I"/<<PKGBUILDDIR>>/GG" -I"/<<PKGBUILDDIR>>/obj-powerpc64le-linux-gnu/GG" -I/usr/include/freetype2 -I"/<<PKGBUILDDIR>>/GG/Boost::boost" -g -O3 -ffile-prefix-map=/<<PKGBUILDDIR>>=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -O2 -g -DNDEBUG -fPIC -fvisibility=hidden -Wall -std=c++17 -MD -MT GG/CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o -MF CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o.d -o CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o -c "/<<PKGBUILDDIR>>/GG/src/dialogs/ColorDlg.cpp"
In file included from /<<PKGBUILDDIR>>/GG/GG/Button.h:22,
                 from /<<PKGBUILDDIR>>/GG/GG/dialogs/ColorDlg.h:18,
                 from /<<PKGBUILDDIR>>/GG/src/dialogs/ColorDlg.cpp:10:
/<<PKGBUILDDIR>>/GG/GG/Base.h:130:16: error: enumerator value ‘-1’ is outside the range of underlying type ‘char’
  130 |     WR_NONE = -1,
      |                ^
/<<PKGBUILDDIR>>/GG/GG/Enum.h:307:9: note: in definition of macro ‘GG_ENUM’
  307 |         __VA_ARGS__                                                                     \
      |         ^~~~~~~~~~~
make[3]: *** [GG/CMakeFiles/GiGi.dir/build.make:79: GG/CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o] Error 1
```

I suppose it happened after https://github.com/freeorion/freeorion/commit/43ff06179bec60704d6b8e7438d30858680b4527 and could be solved with `signed char` definition.

Expected Result
---------------
FreeOrion builds successfully


Steps to reproduce
------------------
<!--
If the error only manifests itself after doing a certain number of actions
please add a list of steps to reproduce the bug.  If the error only occurs in
a certain game state please be sure to add a save game to the issue.

To upload save games please pack them into a Zip archive (*.zip) first.
-->
* Build freeorion of Arm64 or PowerPC 64

--- a/Empire/Diplomacy.h
+++ b/Empire/Diplomacy.h
@@ -20,7 +20,7 @@ FO_ENUM(
 
 class FO_COMMON_API DiplomaticMessage {
 public:
-    enum class Type : char {
+    enum class Type : signed char {
         INVALID = -1,
         WAR_DECLARATION,
         PEACE_PROPOSAL,

--- a/GG/GG/Base.h
+++ b/GG/GG/Base.h
@@ -126,7 +126,7 @@ void ProcessThenRemoveExpiredPtrs(
 
 /** "Regions" of a window; used e.g. to determine direction(s) of drag when a
     window that has a drag-frame is clicked. */
-GG_ENUM(WndRegion, char,
+GG_ENUM(WndRegion, signed char,
     WR_NONE = -1,
     WR_TOPLEFT = 0,
     WR_TOP,

--- a/client/ClientNetworking.h
+++ b/client/ClientNetworking.h
@@ -15,7 +15,7 @@ class Message;
 namespace Networking {
     class AuthRoles;
 
-    enum class RoleType : char;
+    enum class RoleType : unsigned char;
 }
 
 /** Encapsulates the networking facilities of the client.  The client must

--- a/network/Networking.h
+++ b/network/Networking.h
@@ -16,7 +16,7 @@ namespace Networking {
     FO_COMMON_API int DiscoveryPort();
     FO_COMMON_API int MessagePort();
 
-    enum class ClientType : char {
+    enum class ClientType : signed char {
         INVALID_CLIENT_TYPE = -1,
         CLIENT_TYPE_AI_PLAYER,
         CLIENT_TYPE_HUMAN_PLAYER,
@@ -26,7 +26,7 @@ namespace Networking {
     };
     FO_COMMON_API std::ostream& operator<<(std::ostream& os, ClientType client_type);
 
-    enum class RoleType : char {
+    enum class RoleType : unsigned char {
         ROLE_HOST = 0,              ///< allows save and load games, edit other player settings, stop server
         ROLE_CLIENT_TYPE_MODERATOR, ///< allows have a client type Moderator
         ROLE_CLIENT_TYPE_PLAYER,    ///< allows have a client type Player

--- a/parse/ConditionParser.h
+++ b/parse/ConditionParser.h
@@ -10,9 +10,9 @@
 #include <boost/spirit/include/qi.hpp>
 
 namespace Condition {
-    enum class SortingMethod : char;
-    enum class ComparisonType : char;
-    enum class ContentType : char;
+    enum class SortingMethod : unsigned char;
+    enum class ComparisonType : signed char;
+    enum class ContentType : unsigned char;
     struct Condition;
 }
 

--- a/universe/Conditions.h
+++ b/universe/Conditions.h
@@ -24,14 +24,14 @@ namespace ValueRef {
   * Effect system. */
 namespace Condition {
 
-enum class SortingMethod : char {
+enum class SortingMethod : unsigned char {
     SORT_MAX,       ///< Objects with the largest sort key will be selected
     SORT_MIN,       ///< Objects with the smallest sort key will be selected
     SORT_MODE,      ///< Objects with the most common sort key will be selected
     SORT_RANDOM     ///< Objects will be selected randomly, without consideration of property values
 };
 
-enum class ComparisonType : char {
+enum class ComparisonType : signed char {
     INVALID_COMPARISON = -1,
     EQUAL,
     GREATER_THAN,
@@ -41,7 +41,7 @@ enum class ComparisonType : char {
     NOT_EQUAL
 };
 
-enum class ContentType : char {
+enum class ContentType : unsigned char {
     CONTENT_BUILDING,
     CONTENT_SPECIES,
     CONTENT_SHIP_HULL,

--- a/universe/EnumsFwd.h
+++ b/universe/EnumsFwd.h
@@ -1,21 +1,21 @@
 #ifndef _EnumsFwd_h_
 #define _EnumsFwd_h_
 
-enum class UniverseObjectType : char;
-enum class StarType : char;
-enum class PlanetType : char;
-enum class PlanetSize : char;
-enum class PlanetEnvironment : char;
-enum class MeterType : char;
-enum class EmpireAffiliationType : char;
-enum class UnlockableItemType : char;
-enum class ResourceType : char;
-enum class ShipPartClass : char;
-enum class ShipSlotType : char;
-enum class Visibility : char;
-enum class CaptureResult : char;
-enum class ModeratorActionSetting : char;
-enum class FleetAggression : char;
-enum class DiplomaticStatus : char;
+enum class UniverseObjectType : signed char;
+enum class StarType : signed char;
+enum class PlanetType : signed char;
+enum class PlanetSize : signed char;
+enum class PlanetEnvironment : signed char;
+enum class MeterType : signed char;
+enum class EmpireAffiliationType : signed char;
+enum class UnlockableItemType : signed char;
+enum class ResourceType : signed char;
+enum class ShipPartClass : signed char;
+enum class ShipSlotType : signed char;
+enum class Visibility : signed char;
+enum class CaptureResult : signed char;
+enum class ModeratorActionSetting : signed char;
+enum class FleetAggression : signed char;
+enum class DiplomaticStatus : signed char;
 
 #endif

--- a/universe/ValueRef.h
+++ b/universe/ValueRef.h
@@ -78,7 +78,7 @@ std::string FlexibleToString(T&& t)
 [[nodiscard]] FO_COMMON_API std::string FlexibleToString(UniverseObjectType t);
 
 
-enum class ReferenceType : char {
+enum class ReferenceType : signed char {
     INVALID_REFERENCE_TYPE = -1,
     NON_OBJECT_REFERENCE,               // ValueRef::Variable is not evalulated on any specific object
     SOURCE_REFERENCE,                   // ValueRef::Variable is evaluated on the source object

--- a/universe/ValueRefs.h
+++ b/universe/ValueRefs.h
@@ -336,7 +336,7 @@ private:
 
 /** Returns the in-game name of the object / empire / etc. with a specified id. */
 struct FO_COMMON_API NameLookup final : public Variable<std::string> {
-    enum class LookupType : char {
+    enum class LookupType : signed char {
         INVALID_LOOKUP = -1,
         OBJECT_NAME,
         EMPIRE_NAME,

--- a/util/Directories.cpp
+++ b/util/Directories.cpp
@@ -235,7 +235,7 @@ namespace {
     }
 
     constexpr auto PathTypesImpl() {
-        static_assert(std::is_same_v<std::underlying_type_t<PathType>, char>);
+        static_assert(std::is_same_v<std::underlying_type_t<PathType>, signed char>);
         static_assert(char(PathType::PATH_INVALID) > 0);
         std::array<PathType, NUM_PATH_TYPES> retval{};
 

--- a/util/Enum.h
+++ b/util/Enum.h
@@ -27,7 +27,7 @@ enum class \
 BOOST_PP_IF(BOOST_PP_EQUAL(BOOST_PP_TUPLE_SIZE(typeName), 2), \
     BOOST_PP_TUPLE_ELEM(1, typeName), \
     BOOST_PP_TUPLE_ELEM(0, typeName)) \
-: char { \
+: signed char { \
     BOOST_PP_SEQ_FOR_EACH(FO_DEF_ENUM_VALUE, _, values) \
 };
 

--- a/util/GameRules.h
+++ b/util/GameRules.h
@@ -22,7 +22,7 @@ FO_COMMON_API bool RegisterGameRules(GameRulesFn function);
 FO_COMMON_API GameRules& GetGameRules();
 
 struct FO_COMMON_API GameRule : public OptionsDB::Option {
-    enum class Type : char {
+    enum class Type : signed char {
         INVALID = -1,
         TOGGLE,
         INT,

--- a/util/Order.h
+++ b/util/Order.h
@@ -543,7 +543,7 @@ private:
   * \a empire's queue. */
 class FO_COMMON_API ProductionQueueOrder : public Order {
 public:
-    enum class ProdQueueOrderAction : char {
+    enum class ProdQueueOrderAction : signed char {
         INVALID_PROD_QUEUE_ACTION = -1,
         PLACE_IN_QUEUE,
         REMOVE_FROM_QUEUE,

